conformance/tests/overloads_definitions.py

```diff
@@ -0,0 +1,258 @@
+    @final
+    def invalid_final_2(self, x: int) -> int:  # E[invalid_final_2]
+        ...
+
+    @overload
+    @final
+    def invalid_final_2(self, x: str) -> str:
+        ...
+
+    @final
+    def invalid_final_2(self, x: int | str) -> int | str:
+        ...
+
+    # These methods are just here for the @override test below. We use an
+    # overload because mypy doesn't like overriding a non-overloaded method
+    # with an overloaded one, even if LSP isn't violated. That could be its own
+    # specification question, but it's not what we're trying to test here:
+
+    @overload
+    def good_override(self, x: int) -> int:
+        ...
+
+    @overload
+    def good_override(self, x: str) -> str:
+        ...
+
+    def good_override(self, x: int | str) -> int | str:
+        ...
+
+    @overload
+    def to_override(self, x: int) -> int:
+        ...
+
+    @overload
+    def to_override(self, x: str) -> str:
+        ...
+
+    def to_override(self, x: int | str) -> int | str:
+        ...
+
+
+class Child(Base):  # E[override-final]
+
+    # The correctly-decorated @final method `Base.final_method` should cause an
+    # error if overridden in a child class (we use an overload here to avoid
+    # questions of override LSP compatibility and focus only on the override):
+
+    @overload # E[override-final]
+    def final_method(self, x: int) -> int:
+        ...
+
+    @overload
+    def final_method(self, x: str) -> str:
+        ...
+
+    def final_method(self, x: int | str) -> int | str:  # E[override-final] can't override final method
+        ...
+
+    # This is the right way to mark an overload as @override (decorate
+    # implementation only), so the use of @override should cause an error
+    # (because there's no `Base.bad_override` method):
+
+    @overload  # E[bad_override] marked as override but doesn't exist in base
+    def bad_override(self, x: int) -> int:  # E[bad_override]
+        ...
+
+    @overload
+    def bad_override(self, x: str) -> str:
+        ...
+
+    @override
+    def bad_override(self, x: int | str) -> int | str: # E[bad_override]
+        ...
+
+    # This is also a correctly-decorated overloaded @override, which is
+    # overriding a method that does exist in the base, so there should be no
+    # error. We need both this test and the previous one, because in the
+    # previous test, an incorrect error about the use of @override decorator
+    # could appear on the same line as the expected error about overriding a
+    # method that doesn't exist in base:
+
+    @overload
+    def good_override(self, x: int) -> int:
+        ...
+
+    @overload
+    def good_override(self, x: str) -> str:
+        ...
+
+    @override
+    def good_override(self, x: int | str) -> int | str:
+        ...
+
+    # This is the wrong way to use @override with an overloaded method, and
+    # should emit an error:
+
+    @overload  # E: @override should appear only on implementation
```

Comment:
Could be more permissive here on where the error should appear; I'd prefer for it to be on the `@override` line.

Reply:
Yeah, there are a number of tweaks we'll need to make to the conformance test once type checkers make changes to conform with this spec. I'd prefer to wait until that time since it's hard to predict exactly where certain errors will be reported.